src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -55,26 +58,20 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
         {
             viewModel.ThrowIfNull(nameof(viewModel));
 
-            var returnUrl = viewModel.ReturnUrl.ToString();
-            var context = await _interaction.GetAuthorizationContextAsync(returnUrl);
+            var returnUrl = viewModel.ReturnUrl?.ToString();
```

Comment:
There may be a question as to whether we really need to handle the `ReturnUrl == null` case here because the anti-forgery validation essentially limits the invocation of this method to the MVC front-end.

As we set the return URL to the root (in the `GET` method) when the `returnUrl` is `null` we may not need the null-conditional operator.

Reply:
Yeah I agree I don't think we need to worry here about the Null reference. I don't mind belt & braces though just to be sure. Most of the samples I have seen don't seem to worry about it.

Reply:
After thinking about this some more, I'm going to leave the `ReturnUrl == null` checks in as a malicious actor could replay the anti-forgery tokens.